portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +24,26 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  private ArrayList<String> messages = new ArrayList<>();
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    messages.add("Hello!");
+    messages.add("Hola!");
+    messages.add("Hallo!");
+
+    String json = convertToJson(messages);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList messages) {
+    String json = "[";
```

Comment:
Sorry I didn't notice this before: Since `messages` is a list, it'd be cool to iterate over each entry when building this json output, but a full encoder would handle that for you so no point to making this change right now.